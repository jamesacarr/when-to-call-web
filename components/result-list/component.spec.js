/* eslint-disable camelcase */

import React from 'react';
import { shallow } from 'enzyme';

import ResultList from './component';

const results = [
  { place_id: '123', structured_formatting: { main_text: '1', secondary_text: '23' } },
  { place_id: '456', structured_formatting: { main_text: '4', secondary_text: '56' } },
  { place_id: '789', structured_formatting: { main_text: '7', secondary_text: '89' } }
];

const classes = {
  root: 'root'
};

describe('<ResultList/>', () => {
  describe('.render', () => {
    it('renders correctly with result', () => {
      const wrapper = shallow(<ResultList classes={classes} results={results}/>);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders null with empty results', () => {
      const wrapper = shallow(<ResultList classes={classes} results={[]}/>);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders null with null results', () => {
      const wrapper = shallow(<ResultList classes={classes} results={null}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
