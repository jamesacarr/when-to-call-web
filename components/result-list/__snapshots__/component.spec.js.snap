// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultList/> .render renders correctly with result 1`] = `
<WithStyles(Paper)
  className="root"
>
  <WithStyles(List)>
    <ResultRow
      divider={true}
      key="123"
      result={
        Object {
          "place_id": "123",
          "structured_formatting": Object {
            "main_text": "1",
            "secondary_text": "23",
          },
        }
      }
    />
    <ResultRow
      divider={true}
      key="456"
      result={
        Object {
          "place_id": "456",
          "structured_formatting": Object {
            "main_text": "4",
            "secondary_text": "56",
          },
        }
      }
    />
    <ResultRow
      divider={false}
      key="789"
      result={
        Object {
          "place_id": "789",
          "structured_formatting": Object {
            "main_text": "7",
            "secondary_text": "89",
          },
        }
      }
    />
  </WithStyles(List)>
</WithStyles(Paper)>
`;

exports[`<ResultList/> .render renders null with empty results 1`] = `""`;

exports[`<ResultList/> .render renders null with null results 1`] = `""`;
